---
string model
string id
string hardwareVersion
string firmwareVersion
string ipAddress
uint16 port